File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;

    // 가로(width)가 세로(height)보다 크거나 같은 조건으로 탐색
    for (int height = 1; height <= yellow; ++height) {
        if (yellow % height == 0) {
            int width = yellow / height;

            // (가로 + 세로) * 2 + 4 = brown 조건 확인
            if ((width + height) * 2 + 4 == brown) {
                answer.push_back(width + 2); // 가로
                answer.push_back(height + 2); // 세로
                break;
            }
        }
    }

    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int a, b;
        for(int i=yellow ; i>=1 ; i--){
            if(yellow%i==0){
                a=i;
                b=yellow/i;
                
                if(2*a + 2*b +4 ==brown){
                    answer[0] = a + 2;
                    answer[1] = b + 2;
                    return answer;
                }
            }
        }
        return answer;
    }
}